controller/backup_volume_controller.go

```diff
@@ -212,21 +212,18 @@ func (bvc *BackupVolumeController) reconcile(backupVolumeName string) (err error
 
 	log := getLoggerForBackupVolume(bvc.logger, backupVolume)
 
-	// Get default backup target
-	backupTarget, err := bvc.ds.GetBackupTargetRO(types.DefaultBackupTargetName)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
-		}
-		return errors.Wrapf(err, "failed to get %s backup target", types.DefaultBackupTargetName)
+	// Get the backup target of the backup volume
+	backupTarget, err := bvc.ds.GetBackupTargetRO(backupVolume.Spec.BackupTargetName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to get %s backup target", backupVolume.Spec.BackupTargetName)
+	}
+	if backupTarget == nil && backupVolume.DeletionTimestamp == nil {
+		return fmt.Errorf("Failed to find the backup target %v for the backup volume %v", backupVolume.Spec.BackupTargetName, backupVolume.Name)
 	}
 
+	remoteBackupVolumeName := backupVolume.Spec.VolumeName
 	// Examine DeletionTimestamp to determine if object is under deletion
 	if !backupVolume.DeletionTimestamp.IsZero() {
-		if err := bvc.ds.DeleteAllBackupsForBackupVolume(backupVolumeName); err != nil {
-			return errors.Wrap(err, "failed to delete backups")
-		}
-
 		needsCleanupRemoteData, err := checkIfRemoteDataCleanupIsNeeded(backupVolume, backupTarget)
```

Comment:
Why remove `DeleteAllBackupsForBackupVolume`?

Reply:
[The ownerReferences of `Backups`](https://github.com/longhorn/longhorn-manager/pull/2182/files#diff-4967e3520dbda6bc373b2fea83e911dd65aac0ab4f48a53f977413620ce4ba7dR343) will be updated with their corresponding backup volumes so the backups of the backup volume A will be deleted automatically when the backup volume A is deleted.

Reply:
Why did we implement `DeleteAllBackupsForBackupVolume` before?

Reply:
Because we did not add `ownerReferences` into backup volumes and backups CR, we need to delete them in the backup target and backup volume controllers.

Reply:
Adding. Thanks